packages/js/src/ui/components/elements/Preferences/GroupedPreferencesRow.tsx

```diff
@@ -0,0 +1,185 @@
+import { createMemo, createSignal, Index, Show } from 'solid-js';
+
+import { ChannelPreference, ChannelType, Preference } from '../../../../types';
+import { useLocalization } from '../../../context';
+import { useStyle } from '../../../helpers';
+import { Collapsible } from '../../primitives/Collapsible';
+import { ChannelRow } from './ChannelRow';
+import { PreferencesRow } from './PreferencesRow';
+import { Switch, SwitchState } from '../../primitives/Switch';
+import { ArrowDropDown } from '../../../icons/ArrowDropDown';
+import { NodeTree as DefaultNodeTree } from '../../../icons/NodeTree';
+import { Info as DefaultInfo } from '../../../icons/Info';
+import { IconRendererWrapper } from '../../shared/IconRendererWrapper';
+
+export const GroupedPreferencesRow = (props: {
+  group: { name: string; preferences: Preference[] };
+  updatePreference: (preference: Preference) => (channels: ChannelPreference) => void;
+  bulkUpdatePreferences: (preferences: Preference[]) => (channels: ChannelPreference) => void;
+}) => {
+  const style = useStyle();
+  const { t } = useLocalization();
+  const [isOpened, setIsOpened] = createSignal(false);
+
+  const uniqueChannels = createMemo(() => {
+    return props.group.preferences.reduce(
+      (acc, preference) => {
+        Object.keys(preference.channels).forEach((el) => {
+          const channel = el as keyof ChannelPreference;
+          const currentState = acc[channel];
+          const preferenceState = preference.channels[channel] ? 'enabled' : 'disabled';
+          if (!currentState) {
+            acc[channel] = preferenceState;
+          } else {
+            acc[channel] = currentState !== preferenceState ? 'indeterminate' : preferenceState;
+          }
+        });
+
+        return acc;
+      },
+      {} as Record<keyof ChannelPreference, SwitchState>
+    );
+  });
```

Comment:
calculate unique channels for the group, it's used to render group channel toggles